src/integrated/land.jl

```diff
@@ -0,0 +1,669 @@
+        update_canopy_bf!(p, Y, t)
+        # Update soil CO2
+        update_soilco2_bf!(p, Y, t)
+
+        # compute net flux with atmosphere, this is useful for monitoring conservation
+        _LH_f0 = FT(LP.LH_f0(earth_param_set))
+        _ρ_liq = FT(LP.ρ_cloud_liq(earth_param_set))
+        ρe_falling_snow = -_LH_f0 * _ρ_liq # per unit vol of liquid water
+        @. p.atmos_energy_flux =
+            (1 - p.snow.snow_cover_fraction) * (
+                p.soil.turbulent_fluxes.lhf + p.soil.turbulent_fluxes.shf -
+                p.soil.R_n
+            ) +
+            p.snow.snow_cover_fraction * (
+                p.snow.turbulent_fluxes.lhf + p.snow.turbulent_fluxes.shf -
+                p.snow.R_n
+            ) +
+            p.drivers.P_snow * ρe_falling_snow +
+            p.canopy.energy.turbulent_fluxes.shf +
+            p.canopy.energy.turbulent_fluxes.lhf -
+            p.canopy.radiative_transfer.SW_n - p.canopy.radiative_transfer.LW_n
+        @. p.atmos_water_flux =
+            p.drivers.P_snow +
+            p.drivers.P_liq +
+            (1 - p.snow.snow_cover_fraction) * (
+                p.soil.turbulent_fluxes.vapor_flux_liq +
+                p.soil.turbulent_fluxes.vapor_flux_ice
+            ) +
+            p.snow.snow_cover_fraction * p.snow.turbulent_fluxes.vapor_flux +
+            p.canopy.energy.turbulent_fluxes.transpiration
+
+    end
+    return update_boundary_fluxes!
+end
+
+"""
+    lsm_radiant_energy_fluxes!(p,land::LandModel{FT},
+                                canopy_radiation::Canopy.AbstractRadiationModel{FT},
+                                Y,
+                                t,
+                                ) where {FT}
+
+
+A function which computes the net radiation at the ground surface
+give the canopy radiation model, as well as the outgoing radiation,
+and the net canopy radiation.
+
+Returns the correct radiative fluxes for bare ground in the case
+where the canopy LAI is zero. Note also that this serves the role of
+`canopy_radiant_energy_fluxes!`, which computes the net canopy radiation
+when the Canopy is run in standalone mode.
+"""
+function lsm_radiant_energy_fluxes!(
+    p,
+    land::LandModel{FT},
+    canopy_radiation::Canopy.AbstractRadiationModel{FT},
+    Y,
+    t,
+) where {(FT)}
+    canopy = land.canopy
+    canopy_bc = canopy.boundary_conditions
+    radiation = canopy_bc.radiation
+    earth_param_set = canopy.parameters.earth_param_set
+    _σ = LP.Stefan(earth_param_set)
+    LW_d = p.drivers.LW_d
+    SW_d = p.drivers.SW_d
+    c = LP.light_speed(earth_param_set)
+    h = LP.planck_constant(earth_param_set)
+    N_a = LP.avogadro_constant(earth_param_set)
+    (; λ_γ_PAR, λ_γ_NIR) = canopy_radiation.parameters
+    energy_per_photon_PAR = h * c / λ_γ_PAR
+    energy_per_photon_NIR = h * c / λ_γ_NIR
+    T_canopy =
+        ClimaLand.Canopy.canopy_temperature(canopy.energy, canopy, Y, p, t)
+
+    α_soil_PAR = land.soil.parameters.PAR_albedo
```

Comment:
can we get this from soil.parameters in `soil_canopy_model.jl` too?

Reply:
I think so. I made this change.